soh/soh/Enhancements/bossrush.cpp

```diff
@@ -0,0 +1,468 @@
+﻿#include "bossrush.h"
+#include "functions.h"
+#include "macros.h"
+
+#include <array>
+#include <string>
+#include <vector>
+
+extern "C" {
+uint64_t GetUnixTimestamp();
+}
+
+typedef struct BossRushSetting {
+    std::vector<std::string> name;
+    std::vector<std::vector<std::string>> choices;
+} BossRushSetting;
```

Comment:
I'd prefer using `std::array<std::string, N>` here for the different language strings if possible, since we know what the size will be at compile time. Otherwise a bunch of resizing, allocations, and moving of data happens while declaring the array below. i.e.

```suggestion
typedef struct BossRushSetting {
    std::array<std::string, LANGUAGE_MAX> name;
    std::vector<std::array<std::string, LANGUAGE_MAX>> choices;
} BossRushSetting;
```
Or really a CustomMessage struct would be ideal but I haven't made those C-compatible so that would have to be later. Also it's possible that you using an initializer list to initialize the vectors means that they preallocate the correct size to start with and if that's the case then it's fine to have them as vectors imo.